src/UIComponents/Statement.jsx

```diff
@@ -25,20 +38,48 @@ class Statement extends Component {
     return (
       <div style={styles.statement} id="statement">
         Predict{" "}
-        <span style={styles.statementLabel}>
+        <div style={styles.statementLabel}>
           {labelColumn || "..."}
-        </span>
+          {currentPanel === "dataDisplayLabel" && labelColumn && (
+            <div
+              onClick={() => this.removeLabel()}
+              style={styles.statementDeleteIcon}
+            >
+              <div style={styles.statementDeleteCircle} />
+              <div style={styles.statementDeleteX}>
+                <FontAwesomeIcon icon={faTimesCircle} />
+              </div>
+            </div>
+          )}
+        </div>
         <span>
           {" "}
           based on{" "}
-          <span style={styles.statementFeature}>
-            {selectedFeatures.length > 0
-              ? selectedFeatures.join(", ")
-              : "..."}
-          </span>
-          {selectedFeatures.length > 0 && (
-            "."
+          {selectedFeatures.map((selectedFeature, index) => {
+            return (
+              <span key={index}>
+                <div style={styles.statementFeature}>
+                  {selectedFeature}
+                  {currentPanel === "dataDisplayFeatures" && (
+                    <div
+                      onClick={() => this.removeFeature(selectedFeature)}
+                      style={styles.statementDeleteIcon}
+                    >
+                      <div style={styles.statementDeleteCircle} />
+                      <div style={styles.statementDeleteX}>
+                        <FontAwesomeIcon icon={faTimesCircle} />
+                      </div>
+                    </div>
+                  )}
+                </div>
+                {index < selectedFeatures.length - 1 && ", "}
+              </span>
+            );
+          })}
+          {selectedFeatures.length === 0 && (
+            <span style={styles.statementFeature}>...</span>
```

Comment:
Very cool! Just wondering, how come you replaced lines 35-37 with this?

Reply:
Previously we wrapped the whole set of features, separated by commas, in `styles.statementFeature`.  Now, we wrap each individual feature (if there are any), in that style.